common/djangoapps/entitlements/models.py

```diff
@@ -1,10 +1,45 @@
 import uuid as uuid_tools
+from datetime import datetime
 
 from django.conf import settings
+from django.contrib.sites.models import Site
 from django.db import models
+
+from entitlements.helpers import is_entitlement_expired
 from model_utils.models import TimeStampedModel
 
 
+class CourseEntitlementManager(models.Manager):
+    def get_queryset(self):
+        queryset = super(CourseEntitlementManager, self).get_queryset()
+        for entitlement in queryset:
```

Comment:
Will this result in us iterating over the the entire table with every query?

Reply:
Yes, looking into creating a property now that we'll need to use over the default `expired_at` when using the ORM, and will be overriding the GET and LIST methods for the API views. This'll keep the subset we iterate on and updated down to a minimum.